### PR TITLE
Add Universal Segmentation class + mapping

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR adds the `AutoModelForUniversalSegmentation` class and corresponding mapping. Models that can be added to this mapping include DETR, MaskFormer, Mask2Former and OneFormer.

To do:

- [x] update pipeline

@sgugger for some reason make fixup complains:

```
Traceback (most recent call last):
  File "/Users/nielsrogge/Documents/python_projecten/transformers/utils/check_repo.py", line 827, in <module>
    check_repo_quality()
  File "/Users/nielsrogge/Documents/python_projecten/transformers/utils/check_repo.py", line 816, in check_repo_quality
    check_models_are_in_init()
  File "/Users/nielsrogge/Documents/python_projecten/transformers/utils/check_repo.py", line 369, in check_models_are_in_init
    raise Exception(f"The following models should be in the main init: {','.join(models_not_in_init)}.")
Exception: The following models should be in the main init: MaskFormerForUniversalSegmentation.
```

### CODE REVIEW

**Comment by sgugger** [comment]
I think the error message is pretty clear on what to do. The model would also need to be added to the doc page of maskformer if we go through with this.

I'm not convinced we should however. While adding a new auto-model API could make sense (I'd wait to have more than one model though), renaming the model class a year after the model has been released is not something we should do (the same way we keep `GPTLMHeadModel` for instance).

File: src/transformers/models/maskformer/__init__.py [hunk 1]
```diff
@@ -43,7 +43,7 @@
 else:
     _import_structure["modeling_maskformer"] = [
         "MASKFORMER_PRETRAINED_MODEL_ARCHIVE_LIST",
-        "MaskFormerForInstanceSegmentation",
+        "MaskFormerForUniversalSegmentation",
```
Comment (alaradirik) [inline src/transformers/models/maskformer/__init__.py:46]: Shouldn't we keep `MaskFormerForInstanceSegmentation` until it's fully deprecated?

```suggestion
        "MaskFormerForInstanceSegmentation",
        "MaskFormerForUniversalSegmentation",
```

File: src/transformers/models/maskformer/__init__.py [hunk 2]
```diff
@@ -73,7 +73,7 @@
     else:
         from .modeling_maskformer import (
             MASKFORMER_PRETRAINED_MODEL_ARCHIVE_LIST,
-            MaskFormerForInstanceSegmentation,
+            MaskFormerForUniversalSegmentation,
```
Comment (alaradirik) [inline src/transformers/models/maskformer/__init__.py:76]: Same here, shouldn't we keep the old naming import for now?

```suggestion
            MaskFormerForInstanceSegmentation,
            MaskFormerForUniversalSegmentation,
```

**Review by alaradirik — COMMENTED** [review]
Thanks for adding this! Looks good to me but some files still need editing, which cause some tests to fail.

Could you update the references to the deprecated model and output classes in maskformer.mdx, configuration_maskformer.py, image_processing_maskformer.py, test_feature_extraction_maskformer.py, test_modeling_maskformer.py, and update check_repo.py?

**Comment by sgugger** [comment]
I'm unsure why you keep changing more of the `MaskFormerForInstanceSegmentation`. I think I may have been unclear in my previous comments. I am against changing that name in a model that has been around for 10 months now, for the same reason we did not change the name of `GPTLMHeadModel` or `BertLMHeadModel` even if those are not ideal.

File: src/transformers/models/auto/modeling_auto.py [hunk 3]
```diff
@@ -434,6 +434,15 @@
...
+        # MaskFormerForInstanceSegmentation can be removed from this mapping in v5
+        ("maskformer", "MaskFormerForInstanceSegmentation"),
+    ]
+)
+
+MODEL_FOR_UNIVERSAL_SEGMENTATION_MAPPING_NAMES = OrderedDict(
+    [
+        # Model for Universal Segmentation mapping
+        ("detr", "DetrForSegmentation"),
         ("maskformer", "MaskFormerForInstanceSegmentation"),
```
Comment (NielsRogge) [inline src/transformers/models/auto/modeling_auto.py:446]: Ideally all models in this mapping should be called `ForUniversalSegmentation`, but we can keep the name if you prefer so

**Comment by NielsRogge** [comment]
@sgugger I'll keep the old names, but add them to the same (new) mapping.

File: utils/check_repo.py [hunk 4]
```diff
@@ -614,6 +613,7 @@ def find_all_documented_objects():
     "LineByLineTextDataset",
     "LineByLineWithRefDataset",
     "LineByLineWithSOPTextDataset",
+    "MaskFormerForInstanceSegmentation",
```
Comment (sgugger) [inline utils/check_repo.py:616]: To remove, as this is not deprecated anymore.

**Review by sgugger — APPROVED** [review]
Thanks for iterating!

**Pull request merged**